Annex60/BoundaryConditions/WeatherData/ReaderTMY3.mo

```diff
@@ -650,8 +650,14 @@ equation
       "Get HDirNor using user input file";
   elseif  HSou == Annex60.BoundaryConditions.Types.RadiationDataSource.Input_HGloHor_HDifHor then
       Annex60.Utilities.Math.Functions.smoothMin(
-        x1=((HGloHor_in_internal -HDifHor_in_internal)/Annex60.Utilities.Math.Functions.smoothMax(
-        x1=cos(zenAng.zen), x2=epsCos, deltaX=0.1*epsCos)), x2=solCon, deltaX=1e-2)
+        solCon,
+        (HGloHor_in_internal -HDifHor_in_internal)*
+          Annex60.Utilities.Math.Functions.spliceFunction(
+            x=cos(zenAng.zen),
+            pos=Annex60.Utilities.Math.Functions.inverseXRegularized(cos(zenAng.zen),epsCos),
+            neg=0,
+            deltax=epsCos),
+        1e-4 * solCon)
```

Comment:
Please use a constant here such as 0.1 to avoid a multiplication each time the function is invoked.